engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -799,6 +807,10 @@ public void monitorHeartbeat(boolean waitForRecovery)
     processEvents();
 
     committedWindowId = updateCheckpoints(waitForRecovery);
+    if (lastCommittedWindowId != committedWindowId) {
```

Comment:
What happens when there is a stram failure and recovery, is the lastCommitedWindowId also restored?